src/Symfony/Component/String/Utf8String.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\String;
+
+use Symfony\Component\String\Exception\ExceptionInterface;
+use Symfony\Component\String\Exception\InvalidArgumentException;
+
+/**
+ * Represents a string of Unicode code points encoded as UTF-8.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author Hugo Hamon <hugohamon@neuf.fr>
+ *
+ * @throws ExceptionInterface
+ */
+class Utf8String extends AbstractUnicodeString
+{
+    public function __construct(string $string = '')
+    {
+        if ('' !== $string && !preg_match('//u', $string)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        $this->string = $string;
+    }
+
+    public function append(string ...$suffix): AbstractString
+    {
+        $str = clone $this;
+        $str->string .= 1 >= \count($suffix) ? $suffix[0] ?? '' : implode('', $suffix);
+
+        if (!preg_match('//u', $str->string)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        return $str;
+    }
+
+    public function chunk(int $length = 1): array
+    {
+        if (1 > $length) {
+            throw new InvalidArgumentException('The maximum length of each segment must be greater than zero.');
```

Comment:
No, the `$length` parameter is the maximum length of each chunk. The last chunk can be smaller.

Reply:
@gharlan but there's  `1 > $length`, it confused me a bit. Maybe it should be just "The length of each segment ..."?

Reply:
I think the current message is correct, but I agree, it can be confusing.
We could use this instead:

> The length parameter must be greater than zero.

or this:

> Argument 1 must be greater than zero.

Reply:
now using `The chunk length must be greater than zero.`